irksome/stage_value.py

```diff
@@ -236,6 +253,11 @@ def _update_general(self):
         self.update_solver.solve()
         self.u0.assign(self.unew)
 
+    def _update_collocation(self):
+        stage_vals = self.u0.subfunctions + self.stages.subfunctions
+        for i, u0bit in enumerate(self.u0.subfunctions):
+            u0bit.assign(numpy.dot(self.collocation_vander[:], stage_vals[i::self.num_fields]))
```

Comment:
Maybe this is the fix you need
```suggestion
            u0bit.assign(numpy.dot(stage_vals[i::self.num_fields]. self.collocation_vander))
```

Reply:
This should be fixed now